src/components/SigningBox/index.tsx

```diff
@@ -1,12 +1,27 @@
 import { FC, useState, useEffect } from 'react';
 import { motion, AnimatePresence } from 'motion/react';
+import { create } from 'zustand';
 
 import accountBalanceWalletIcon from '../../assets/account-balance-wallet-blue.svg';
 import { OfframpSigningPhase } from '../../types/offramp';
 import { isNetworkEVM } from '../../helpers/networks';
 import { useNetwork } from '../../contexts/network';
 import { Spinner } from '../Spinner';
 
+interface SignatureState {
+  max: number;
+  current: number;
+  setSigners: (max: number, current: number) => void;
+  reset: () => void;
+}
+
+export const useSignatureStore = create<SignatureState>((set) => ({
```

Comment:
Since this is not only used in this component but also in the safe-wallet hooks, maybe it makes sense to move it into the `store/` directory?